src/jit/lower.cpp

```diff
@@ -3685,6 +3685,9 @@ void Lowering::DoPhase()
         {
             TreeNodeInfoInit(node);
 
+            // Only nodes that produce values should have a non-zero dstCount.
+            assert((node->gtLsraInfo.dstCount == 0) || node->IsValue());
+
```

Comment:
Curiously, why should normalized form be avoided in this case; as the ANF of `(X && !Y) || (!X && Y)` is `X || Y`?